jobs/jobDbRepos.go

```diff
@@ -1,31 +1,147 @@
 package jobs
 
-import "github.com/SaiNageswarS/go-api-boot/odm"
+import (
+	"time"
 
-type JobDetailRepository struct {
-	odm.AbstractRepository[JobDetailModel]
-}
+	"github.com/SaiNageswarS/go-api-boot/odm"
+)
 
 type JobRunLogRepository struct {
-	odm.AbstractRepository[JonRunLogModel]
+	odm.AbstractRepository[JobRunLogModel]
 }
 
 type JobDb struct {
 	Database string
 }
 
-func (j *JobDb) GetJobDetailRepository() *JobDetailRepository {
-	return &JobDetailRepository{
-		AbstractRepository: odm.AbstractRepository[JobDetailModel]{
-			Database:       j.Database,
-			CollectionName: "JobDetails",
-		},
-	}
+func (j *JobRunLogRepository) IsRunning(id string) chan bool {
+	ch := make(chan bool)
+
+	go func() {
+		resultChan, errorChan := j.FindOneById(id)
+
+		select {
+		case res := <-resultChan:
+			ch <- res.LastExecutionStatus == JobStatusRunning
+		case _ = <-errorChan:
+			ch <- false
+		}
+	}()
+
+	return ch
+}
+
+func (j *JobRunLogRepository) MarkJobAsRunning(id string) chan error {
+	ch := make(chan error)
```

Comment:
Instead of having a separate method for each status of markJob, can we have a single method and pass status as parameter?